src/App.js

```diff
@@ -87,252 +96,266 @@ class Root extends React.Component {
   if (routeRequiresWalletToBeLoaded) {
-    // Wallet is not loaded or loading, but it is started
-    // Since it requires the wallet to be loaded, redirect to the wallet_type screen
     return <Redirect to={{pathname: '/wallet_type/'}}/>;
   }
-  // Wallet is not loaded nor loading, and the route does not require it.
-  // Do not redirect anywhere, just render the component.
-  return <Component {...props} />;
-};
 
-/*
- * Route for the components that will be shown after the wallet was started (After user clicked in 'Get started' in Welcome screen)
- */
-const StartedRoute = ({component: Component, ...rest}) => (
-  <Route {...rest} render={(props) => (
-    returnStartedRoute(Component, props, rest)
-  )} />
-)
-
-/*
- * Return a div grouping the Navigation and the Component
- */
-const returnDefaultComponent = (Component, props) => {
-  const reduxState = store.getState();
+  // Wallet is not loaded nor loading, and the route does not require it to be loaded.
+  // Do not redirect anywhere: just render the component.
+  return children;
+}
 
-  if (reduxState.isVersionAllowed === undefined) {
+/**
+ * Renders the selected children component, wrapping it with the navigation bar on top and the error modal.
+ * It also handles the version checking, redirecting to an error screen if the wallet version is obsolete.
+ * @param children
+ * @returns {JSX.Element}
+ * @constructor
+ */
+function DefaultComponent({ children }) {
+  const { isVersionAllowed } = useSelector(state => ({
+    isVersionAllowed: state.isVersionAllowed,
+  }));
+  const history = useHistory();
 
-    helpers.loadStorageState();
+  const [versionIsKnown, setVersionIsKnown] = useState(false);
 
-    // We already handle all js errors in general and open an error modal to the user
-    // so there is no need to catch the promise error below
-    version.checkApiVersion(reduxState.wallet);
-  }
-
-  if (version.checkWalletVersion()) {
-    if (
-      props.location.pathname === '/locked/' &&
-      LOCAL_STORE.isHardwareWallet()
-    ) {
-      // This will redirect the page to Wallet Type screen
-      LOCAL_STORE.cleanWallet();
-      return <Redirect to={{ pathname: '/wallet_type/' }} />;
-    } else {
-      return (
-        <div className='component-div h-100'>
-          <Navigation />
-          <Component {...props} />
-          <RequestErrorModal />
-        </div>
-      );
+  // Monitors the version data from state and allows the rendering of the screen accordingly
+  useEffect(() => {
+    if (isVersionAllowed === undefined) {
+      // The version information is not yet available, no op
+      return;
+    }
+    setVersionIsKnown(true);
+  }, [isVersionAllowed]);
+
+  // Do not any wallet screens if the wallet version is still unknown. Only the Locked Sreen is allowed
+  const isLockedScreen = history.location.pathname === '/locked/';
+  if (!versionIsKnown) {
+    if (!isLockedScreen) {
+      return null;
     }
```

Comment:
I didn't understand it. Shouldn't it return at least an error page?

Reply:
This indeed is still confusing. I believe the outer navigation and outer wrappers should be refactored in a future moment for increased legibility.

Added some more documentation on [ae60eeb](https://github.com/HathorNetwork/hathor-wallet/pull/502/commits/ae60eebbab367a2be5c3dba0c28b41c5149bcfd6), but in short: this screen should not render anything while the version is still unknown.

The response from the fullnode informing the version should be received in a fraction of a second, so this will normally not be an issue. Connection failures with the fullnode during this request should be intercepted by the `<ErrorWrapper />` component